### PR TITLE
Add email format check for sign in

### PR DESCRIPTION
**Pull request by xswoodsx**
## Ticket and context

Ticket: [921](https://dfedigital.atlassian.net/jira/software/projects/CPDRP/boards/102?assignee=5f3543bb3e9e2e004db3189d&selectedIssue=CPDRP-921)

### Is there anything that the code reviewer should know?

The regex in Devise initializer hasn't been used as it didn't cover many invalid formats i tried, e.g "invalid@email,com"

### Code quality checks
- [ ] All commit messages are meaningful and true
- [ ] Added enough automated tests

### HTML Checks
- [ ] All new pages have automated accessibility checks
- [ ] All new pages have visual tests via Percy

### Gotchas
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Try to log in with an incorrect email format 

## External API changes

Does this change anything in our external APIs? If so, did you remember to update the CHANGELOG for Lead Providers? (lead-providers/guidance/release-notes)

https://user-images.githubusercontent.com/69353998/135220212-c39613f8-92d5-46fd-99ca-2f6f254e9437.mov

### CODE REVIEW

File: app/controllers/users/sessions_controller.rb [hunk 1]
```diff
@@ -86,4 +91,29 @@ def mock_login
...
+
+    user = User.find_by(email: params[:user][:email])
+
+    if user.blank?
+      user = User.new(email: email)
+      user
+    end
+
+    user
+  end
```
Comment (kicferk1) [inline app/controllers/users/sessions_controller.rb:118]: There is a rather more sophisticated email format checker in our code already. `NotifyEmailValidator`.

You use it by making a model, and in that model, writing something like
```
  validates :email, notify_email: true
```

I wonder if we want to make a form object for sign in to wire it up nicely.

Comment (ethanmills) [reply app/controllers/users/sessions_controller.rb:118]: We're using `User` here, which already has ` validates :email, presence: true, uniqueness: true, notify_email: true`, so we probably hijack that

Comment (ethanmills) [reply app/controllers/users/sessions_controller.rb:118]: Although maybe not, since that would cause issues with uniqueness

Comment (kicferk1) [reply app/controllers/users/sessions_controller.rb:118]: Oh, interesting, that would be even nicer :)

**Review by kicferk1 — APPROVED** [review]

**Review by ethanmills — APPROVED** [review]

**Pull request merged**